rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+  /v1/namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: boolean
+        # TODO - Not empty exception, No such namespace exception
+        '401':
+          description: Unauthorized
+components:
+  schemas:
+    CommitTableRequest:
+      type: object
+      properties:
+        tableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        metadataJson:
+          type: string
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+      properties:
+        namespace:
+          type: array
+          items:
+            type: string
+          nullable: false  # Note this is ignored
+        name:
+          pattern: \S
+          type: string
+          nullable: false
+    CommitTableResponse:
+      type: object
+      properties:
+        metadataLocation:
+          type: string
+        metadataJson:
+          type: string
+    RemovePropertiesRequest:
+      type: object
+      properties:
+        namespace:
+          type: string
+        properties:
+          uniqueItems: true
+          type: array
+          items:
+            type: string
+    Catalog:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: Unique identifier for this catalog
+        name:
+          type: string
+        location:
+          type: string
+          description: Warehouse location for this catalog or URI of metastore or other identifying location
+        properties:
+          type: object
+          description: Additional catalog level properties
+          default: {}
+      required:
+        - name
+        - properties
+    CreateNamespaceRequest:
+      type: object
+      properties:
+        namespace:
+          type: array
+          description: individual levels of the namespace
+          items:
+            type: string
+        properties:
+          type: object
+          description: Configuration properties for the namespace
+          example: '{ owner: "Hank Bendickson" }'
```

Comment:
:smile:

Reply:
Should the example be a valid JSON object?

Reply:
Oh yeah I can add the single quotes. In javascript, that would be accepted as a JSON object. So I meant for that to be a JSON object, but made the keys optionally strings as most JSON parsers allow that.

In the actual implementation, we'd use strings just to be sure. I can update that.